### PR TITLE
Ims maintenance endpoints

### PR DESCRIPTION
**Pull request by MatteoGuarnaccia5**
## Description

see https://github.com/ral-facilities/ldap-jwt-auth/issues/88

## Testing instructions

- [ ] Review code
- [ ] Check Actions build
- [ ] Review changes to test coverage

### CODE REVIEW

File: src/authentication/ldapjwtAuthProvider.test.tsx
Comment (joelvdavies) [inline src/authentication/ldapjwtAuthProvider.test.tsx:1]: Filename should probably be ldapJWTAuthProvider, same for others.

File: src/authentication/ldapjwtAuthProvider.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import { MaintenanceState } from '../state/scigateway.types';
+
+export default class LDAPJWTAuthProvider extends JWTAuthProvider {
+  public fetchScheduledMaintenanceState(): Promise<ScheduledMaintenanceState> {
+    return Axios.get(`${this.authUrl}/scheduled_maintenance`)
+      .then((res) => {
+        return res.data;
+      })
+      .catch((err) => {
+        this.handleAuthError(err);
```
Comment (joelvdavies) [inline src/authentication/ldapjwtAuthProvider.tsx:13]: The codecov warnings should be fixed, see `should log the user out if it fails to fetch scheduled maintenance state` in the icatAuth tests (same for the one below). I know you mentioned about whether it should do that but looking at it in more detail it only logs out when the error code is 403 or 401 so I think the test name is misleading but it also covers this line at the same time.

File: src/state/reducers/scigateway.reducer.tsx [hunk 2]
```diff
@@ -365,6 +366,10 @@ export function handleAuthProviderUpdate(
         provider = new JWTAuthProvider(payload.authUrl);
         break;
 
+      case 'ldapjwt':
```
Comment (joelvdavies) [inline src/state/reducers/scigateway.reducer.tsx:369]:
```suggestion
      case 'ldap-jwt':
```
The codecov warning here should also be fixed.

**Review by joelvdavies — COMMENTED** [review]
A couple minor comments, works with https://github.com/ral-facilities/ldap-jwt-auth/pull/96 though.

File: src/state/middleware/scigateway.middleware.tsx [hunk 3]
```diff
@@ -171,9 +171,9 @@ export const listenToPlugins = (
             if (pluginMessage.detail.payload.severity !== undefined) {
               const { severity, message } = pluginMessage.detail.payload;
 
-              if (severity !== 'success') {
-                toastrMessage(message, severity);
-              }
+              // if (severity !== 'success') {
+              toastrMessage(message, severity);
+              //}
```
Comment (joelvdavies) [inline src/state/middleware/scigateway.middleware.tsx:176]:
```suggestion
              if (severity !== 'success') {
                  toastrMessage(message, severity);
              }
```

File: src/authentication/ldapjwtAuthProvider.test.tsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
+import mockAxios from 'axios';
+import LDAPJWTAuthProvider from './ldapjwtAuthProvider';
+
+describe('ldapJWT auth provider', () => {
+  let ldapJjwtAuthProvider: LDAPJWTAuthProvider;
+
+  beforeEach(() => {
+    ldapJjwtAuthProvider = new LDAPJWTAuthProvider('http://localhost:8000');
```
Comment (joelvdavies) [inline src/authentication/ldapjwtAuthProvider.test.tsx:8]:
```suggestion
    ldapJWTAuthProvider = new LDAPJWTAuthProvider('http://localhost:8000');
```

File: src/authentication/ldapjwtAuthProvider.test.tsx [hunk 5]
```diff
@@ -0,0 +1,75 @@
+import mockAxios from 'axios';
+import LDAPJWTAuthProvider from './ldapjwtAuthProvider';
+
+describe('ldapJWT auth provider', () => {
```
Comment (joelvdavies) [inline src/authentication/ldapjwtAuthProvider.test.tsx:4]:
```suggestion
describe('LDAP-JWT Auth provider', () => {
```

File: src/state/middleware/scigateway.middleware.tsx (hunk 3 above)
Comment (MatteoGuarnaccia5) [reply src/state/middleware/scigateway.middleware.tsx:176]: was not meant to commit this change sorry

File: src/state/reducers/scigateway.reducer.test.tsx [hunk 6]
```diff
@@ -280,6 +281,12 @@ describe('scigateway reducer', () => {
       ICATAuthProvider
     );
 
+    updatedState = ScigatewayReducer(state, loadAuthProvider('ldap-jwt'));
+
+    expect(updatedState.authorisation.provider).toBeInstanceOf(
+      LDAPJWTAuthProvider
+    );
```
Comment (joelvdavies) [inline src/state/reducers/scigateway.reducer.test.tsx:288]: Could this be moved after the icat.anon one as its the same as the icat one above.

**Review by joelvdavies — APPROVED** [review]
Merge this at the same time as the ldap-jwt-auth PR and docker-ims MR.

**Pull request merged**